core/chains/evm/client/pool.go

```diff
@@ -195,36 +196,41 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
-	nodes := p.liveNodes()
-	nNodes := len(nodes)
-	if nNodes == 0 {
-		p.logger.Critical("No live RPC nodes available")
-		return &erroringNode{errMsg: fmt.Sprintf("no live nodes available for chain %s", p.chainID.String())}
+func (p *Pool) getBestNode() Node {
+	highestHeadNumber := int64(-1)
+	var node Node
```

Comment:
maybe call this bestNode?

Reply:
Just followed the previous naming (was getRoundRobin).